src/__tests__/user.spec.js

```diff
@@ -0,0 +1,184 @@
+import chai from 'chai';
+import chaiHttp from 'chai-http';
+import sinon from 'sinon';
+import 'chai/register-should';
+
+import app from '../index';
+import { userId, wrongId, updateUser, validToken, inValidToken2, invalidUserData, inValidRequest, invalidSyntax, updateUser2 } from './mocks/userMock'
+chai.use(chaiHttp);
+const { expect } = chai;
+
+const BASE_URL = '/api/v1';
+
+let request;
+
+describe('User Profile Route', () => {
```

Comment:
Include a hook that prepares the database for your test and clears it after the test is completed

Reply:
why do we have to do this? 
Is it really necessary? if Yes
Can you show me HOW to implement that?

Reply:
Reason is so that you expectation for a seeded database is not compromised by some other tests that have conflicting dataset or have cleared the database after it's test. You want to make sure you have control over your own sandboxed environment and deliver the workspace afresh after your test is done, for other members test to proceed on a clean slate.

I implemented something like this in this way

`
// clear database and seed data before test
before(async () => {
    try {
        await clearTestDb();
        await seedTestDb();
    } catch (err) {
        throw err;
    }
});

// clear database after test
after(async () => {
    try {
        await clearTestDb();
    } catch (err) {
        throw err;
    }
});
`

Reply:
`// Clear tables of seed
const clearTestDb = async () => {
    try {
        await models['User'].destroy({ where: {}, force: true });
        await models['Login'].destroy({ where: {}, force: true });
        await models['Reset'].destroy({ where: {}, force: true });
    } catch (err) {
        throw err;
    }
};`

Reply:
// Create table and seed database
const seedTestDb = async () => {
    await models.User.create({
        email: 'youremail3@andela.com',
    });

    await models.User.create({
        email: 'youremail4@andela.com',
    });

    await models.Login.create({
        email: 'youremail3@andela.com',
        password: 'password'
    });

    await models.Login.create({
        email: 'youremail4@andela.com',
        password: 'password'
    });

    await models.Reset.create({});
};

Reply:
@conquext Yea, that's a valid reason.
What about the case whereby I have userMock file that has the format of what the database should like and for the case of creating tables and seeding data, can we have a `script` that create and drop table?

Reply:
yes we can, the one i posted is an example. if you are able to mock the database that is also fine.

Reply:
okay, I will stick with mocking user's data